### ISSUE TITLE
Importing @prisma/client takes 6 seconds

### ISSUE THREAD

**Issue by flesler** [prisma/prisma#8178]
### Bug description

I first asked via Slack here: https://prisma.slack.com/archives/CA491RJH0/p1625770401309600

This exact line takes 6s in my project: `import { PrismaClient } from '@prisma/client'`

### How to reproduce

- Clone this repo: github.com/flesler/prisma-boot-test
- `npm install`
- `npm run build`
- `npm start`

For me, it takes 4.5s (as shown in the log) even though all it does is import Prisma.

I timed different bits of the generated JS code until I found the culprit at this:

```
const dirname = findSync(process.cwd(), [
  '"node_modules/.prisma/client"',
  '".prisma/client"',
], ['d'], ['d'], 1)[0] || __dirname
```

### Expected behavior

Import time should be just some milliseconds

### Prisma information

The schema from your starter already reproduces it (as shown in the link above)

### Environment & setup

- OS: Ubuntu 20
- Database: any, happens with PostgreSQL 10 and SQLite
- Node.js version: 14.17.0

### Prisma Version

```
$ ./node_modules/.bin/prisma -v
Environment variables loaded from .env
prisma               : 2.26.0
@prisma/client       : 2.26.0
Current platform     : debian-openssl-1.1.x
Query Engine         : query-engine 9b816b3aa13cc270074f172f30d6eda8a8ce867d (at node_modules/@prisma/engines/query-engine-debian-openssl-1.1.x)
Migration Engine     : migration-engine-cli 9b816b3aa13cc270074f172f30d6eda8a8ce867d (at node_modules/@prisma/engines/migration-engine-debian-openssl-1.1.x)
Introspection Engine : introspection-core 9b816b3aa13cc270074f172f30d6eda8a8ce867d (at node_modules/@prisma/engines/introspection-engine-debian-openssl-1.1.x)
Format Binary        : prisma-fmt 9b816b3aa13cc270074f172f30d6eda8a8ce867d (at node_modules/@prisma/engines/prisma-fmt-debian-openssl-1.1.x)
Default Engines Hash : 9b816b3aa13cc270074f172f30d6eda8a8ce867d
Studio               : 0.408.0
```

**Comment by janpio** [comment]
This is a cross effect of your dependency tree, if you remove the dependencies (and remove `node_modules`, run `npm install`), the duration goes down to under 1s. Still not great, but perfectly fine usually.

We'll need to investigate here why this side effect happens and what we can do to improve this.

**Comment by flesler** [comment]
Right, obviously. But those dependencies are needed and, this being a young project, the list is not even that large. I think you need to rewrite the `findSync` asap to do what you need it to do without (as it seems) having to crawl node_modules all over the place

**Comment by Saeid-Za** [comment]
Same issue here: #7457
@flesler thanks for investigating this a bit further.
Did you try to replace `findSync` with somthing else and remeasure the load time?

**Comment by flesler** [comment]
@Saeid-Za yes, changing it to `const dirname = __dirname` obviously solves the problem. I made a script that patches the generated JS for now after each `prisma generate` :smile:

**Comment by pantharshit00** [comment]
Issue is reproducible and sounds like an easy fix. Marking as candidate.

**Comment by flesler** [comment]
@janpio @pantharshit00 @matthewmueller  on my project (without my patch) this is now at >12s and we haven't reached production yet. The time skyrockets exponentially when one adds dependencies.

I think this is much bigger issue than you might be feeling, it's just that most people never figure out Prisma is the reason their projects are taking forever to boot and never complain here.

**Comment by jderrough** [comment]
@flesler How did you hook up your patch to run when the client is generated?

I'm building a Redwoodjs app, so I'm not in total control of the when `prisma generate` is executed but this is getting really annoying. At the moment, I have 32 test files, all importing '@prisma/client'. The tests take about 4 min to run on my computer, about 3/4 of which is prisma looking up for the same folder over and over. And it's only getting worse as the project grows.

**Comment by flesler** [comment]
@jderrough in `package.json`:

```
"scripts": {
  "ts": "ts-node --transpile-only",
  "db:format": "prisma format",
  "predb:migrate": "npm run db:format",
  "db:migrate": "npm run prisma -- migrate dev --skip-generate",
  "postdb:migrate": "npm run db:generate",
  "db:generate": "prisma generate",
  "postdb:generate": "npm run ts bin/patch-prisma-client",
}
```

**Comment by jderrough** [comment]
Thanks for sharing and for raising this issue, @flesler.

Unfortunately for me, `prisma generate` is called by the Redwoodjs toolchain and it doesn't look like I can run a patch every time it is done.

But hopefully, this will be addressed in prisma soon. 🤞🏼

**Comment by janpio** [comment]
This is actively being worked on now. No promises it will make the next release, but expect us to definitely look into this and see how/if/when we can fix this.

**Comment by flesler** [comment]
I (the OP) just upgraded to 3.4.0 and re-tested. I can confirm the boot time now is the same as it was in older versions with my patch :ok_hand:
